src/module-init/module-init.c

```diff
@@ -141,6 +142,11 @@ void RS_moduleInfoFunc(RedisModuleInfoCtx *ctx, int for_crash_report) {
   RedisModule_InfoAddFieldDouble(ctx, "total_cycles", stats.totalCycles);
   RedisModule_InfoAddFieldDouble(ctx, "total_ms_run", stats.totalTime);
 
+  // Query statistics
+  RedisModule_InfoAddSection(ctx, "queries");
+  RedisModule_InfoAddFieldLongLong(ctx, "total_queries", RSGlobalStats.totalStats.total_unique_queries);
+  RedisModule_InfoAddFieldLongLong(ctx, "total_query_commands", RSGlobalStats.totalStats.total_query_commands);
+
```

Comment:
It can be atomic. IMO the exact number doesn't matter much when reading it (when counting, we don't want to miss counts so it has to be atomic).
If we want the read to be atomic, we should have an API that returns a copy of the stats struct or something

Reply:
The problem is that the result of unsafe access to atomic value can be garbage

Reply:
Is it still true in 2024?